hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AdjustAllowanceChecks.java

```diff
@@ -65,32 +70,46 @@
 	public AdjustAllowanceChecks(
 			final Supplier<MerkleMap<EntityNumPair, MerkleUniqueToken>> nftsMap,
 			final TypedTokenStore tokenStore,
+			final AccountStore accountStore,
 			final GlobalDynamicProperties dynamicProperties) {
+		this.accountStore = accountStore;
 		this.tokenStore = tokenStore;
 		this.nftsMap = nftsMap;
 		this.dynamicProperties = dynamicProperties;
 	}
 
 	@Override
 	public ResponseCodeEnum validateCryptoAllowances(final List<CryptoAllowance> cryptoAllowancesList,
-			final Account ownerAccount) {
+			final Account payerAccount) {
 		if (cryptoAllowancesList.isEmpty()) {
 			return OK;
 		}
 		final var cryptoKeys = cryptoAllowancesList
 				.stream()
-				.map(CryptoAllowance::getSpender)
+				.map(allowance -> buildEntityNumPairFrom(
```

Comment:
Streams are fun, but we should avoid them in `handleTransaction()` pretty much always. 😞